test/functional/api/v2/discovered_hosts_controller_test.rb

```diff
@@ -29,6 +29,9 @@ def switch_controller(klass)
                        :name => 'discovery_reboot',
                        :value => true,
                        :category => 'Setting::Discovered')
+
+    #load the power.rb file, as PowerService is not autoloaded from file named power
+    ForemanDiscovery::NodeAPI::Power 
```

Comment:
Please split it, good catch. My fault :-( Would you mind doing this in a separate patch? This might be problematic in the future and we might want to backport this.

Looks great, aside of Katello, this was the most complicated plugin! Well done!